src/transformers/models/grounding_dino/processing_grounding_dino.py

```diff
@@ -0,0 +1,209 @@
+                pad_to_multiple_of=pad_to_multiple_of,
+                return_attention_mask=return_attention_mask,
+                return_overflowing_tokens=return_overflowing_tokens,
+                return_special_tokens_mask=return_special_tokens_mask,
+                return_offsets_mapping=return_offsets_mapping,
+                return_token_type_ids=return_token_type_ids,
+                return_length=return_length,
+                verbose=verbose,
+                return_tensors=return_tensors,
+                **kwargs,
+            )
+        else:
+            text_encoding = None
+
+        if text_encoding is not None:
+            # Keeping same order of model_input_names when both images and text
+            text_encoding.update(encoding_image_processor)
+            encoding_image_processor = text_encoding
+
+        return encoding_image_processor
+
+    # Copied from transformers.models.blip.processing_blip.BlipProcessor.batch_decode with BertTokenizerFast->PreTrainedTokenizer
+    def batch_decode(self, *args, **kwargs):
+        """
+        This method forwards all its arguments to PreTrainedTokenizer's [`~PreTrainedTokenizer.batch_decode`]. Please
+        refer to the docstring of this method for more information.
+        """
+        return self.tokenizer.batch_decode(*args, **kwargs)
+
+    # Copied from transformers.models.blip.processing_blip.BlipProcessor.decode with BertTokenizerFast->PreTrainedTokenizer
+    def decode(self, *args, **kwargs):
+        """
+        This method forwards all its arguments to PreTrainedTokenizer's [`~PreTrainedTokenizer.decode`]. Please refer to
+        the docstring of this method for more information.
+        """
+        return self.tokenizer.decode(*args, **kwargs)
+
+    @property
+    # Copied from transformers.models.blip.processing_blip.BlipProcessor.model_input_names
+    def model_input_names(self):
+        tokenizer_input_names = self.tokenizer.model_input_names
+        image_processor_input_names = self.image_processor.model_input_names
+        return list(dict.fromkeys(tokenizer_input_names + image_processor_input_names))
+
+    def post_process_grounded_object_detection(
+        self,
+        outputs,
+        input_ids,
+        box_threshold: float = 0.25,
+        text_threshold: float = 0.25,
+        target_sizes: Union[TensorType, List[Tuple]] = None,
+    ):
+        """
+        Post-process the output of the model to get the grounded object detection results.
+        """
+        results = self.image_processor.post_process_object_detection(outputs, box_threshold, target_sizes)
+
+        probs = torch.sigmoid(outputs.logits)  # (batch_size, num_queries, 256)
+
+        for idx, (result, prob) in enumerate(zip(results, probs)):
+            labels = result["labels"]
+            # Assuming that selected bboxes are sorted by confidence due to Hungarian matching loss in training
+            prob = prob[: len(labels)]  # len(labels) , 256
```

Comment:
I did a re-write of this method to avoid having to make this assumption here https://github.com/huggingface/transformers/pull/26087/commits/09ae5c1708be994896cf8dc791abd3b0bac71e5d